adcc/misc.py

```diff
@@ -47,6 +48,35 @@ def get(self):
     return property(get)
 
 
+def cached_function(function):
```

Comment:
Good idea, but what I don't like about it is that this way the cache is not really easily accessible, which could be an issue in the future. I have to think about this a bit more.

Reply:
Sure, you can just `print(mp.t2.cache)` the way I implemented this.

Reply:
Well ok, but what I meant is that there is no "central" place to manage the cache, but then otoh you can just loop over the cached functions from lazymp somehow. Hmm. Not sure yet if I like it or not at the moment. Just leave it for now and we'll see.

Reply:
I have now implemented a "central" cache 😄